File: TP1/Exo 1.8/C/calculs.c
#include <stdio.h> // Inclusion de la bibliothèque standard pour les entrées/sorties

// Fonction principale
int main() {
    int num1, num2, result; // Déclaration de variables pour stocker les deux nombres et le résultat
    char op; // Variable pour stocker l'opérateur

    // Lecture des entrées utilisateur
    printf("Entrez le premier nombre : ");
    scanf("%d", &num1); // Lecture du premier entier
    printf("Entrez le second nombre : ");
    scanf("%d", &num2); // Lecture du second entier
    printf("Entrez l'opérateur (+, -, *, /, %%, &, |, ~) : ");
    scanf(" %c", &op); // Lecture de l'opérateur (le caractère `%` est échappé par `%%` dans scanf)

    // Utilisation d'un switch pour déterminer l'opération en fonction de l'opérateur
    switch (op) {
        case '+':
            // Addition
            result = num1 + num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '-':
            // Soustraction
            result = num1 - num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '*':
            // Multiplication
            result = num1 * num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '/':
            // Division
            if (num2 == 0) {
                // Gestion de la division par zéro
                printf("Erreur : Division par zéro\n");
            } else {
                result = num1 / num2;
                printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            }
            break;
        case '%':
            // Modulo (reste de la division)
            if (num2 == 0) {
                // Gestion de la division par zéro
                printf("Erreur : Division par zéro\n");
            } else {
                result = num1 % num2;
                printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            }
            break;
        case '&':
            // Opération AND bit à bit
            result = num1 & num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '|':
            // Opération OR bit à bit
            result = num1 | num2;
            printf("Résultat de %d %c %d = %d\n", num1, op, num2, result);
            break;
        case '~':
            // Opération NOT bit à bit sur num1 seulement (num2 est ignoré)
            result = ~num1;
            printf("Résultat de ~%d = %d\n", num1, result);
            break;
        default:
            // Gestion d'un opérateur non reconnu
            printf("Erreur : Opérateur non reconnu '%c'\n", op);
            break;
    }

    return 0; // Fin du programme
}

File: TP1/Exo 1.8/Python/calculs.py
# calculs.py

def calculer(a, b, op):
    # Fonction pour effectuer des calculs de base
    try:
        # Utilisation de if-elif pour déterminer l'opération à effectuer
        if op == '+':
            return a + b
        elif op == '-':
            return a - b
        elif op == '*':
            return a * b
        elif op == '/':
            # Gestion de la division par zéro
            if b == 0:
                raise ValueError("Erreur : Division par zéro")
            return a / b
        elif op == '%':
            # Gestion de la division par zéro
            if b == 0:
                raise ValueError("Erreur : Division par zéro")
            return a % b
        else:
            # Gestion de l'opérateur non reconnu
            raise ValueError(f"Erreur : Opérateur non reconnu '{op}'")
    except ValueError as e:
        # Capture des erreurs de type ValueError et retourne le message d'erreur
        return str(e)

def main():
    # Lecture des entrées utilisateur
    try:
        a = float(input("Entrez le premier nombre : "))  # Lecture du premier nombre
        b = float(input("Entrez le second nombre : "))   # Lecture du second nombre
        op = input("Entrez l'opérateur (+, -, *, /, %) : ")  # Lecture de l'opérateur

        # Calcul et affichage du résultat
        resultat = calculer(a, b, op)
        print(f"Le résultat de {a} {op} {b} est : {resultat}")

    except ValueError as e:
        # Gestion des erreurs lors de la saisie des nombres
        print(f"Erreur de saisie : {e}")

# Point d'entrée du programme
if __name__ == "__main__":
    main()
